src/Angor/Client/Pages/Signatures.razor

```diff
@@ -161,6 +175,40 @@
         }
     }
 
+    private async Task RefreshSignaturesInternal()
+    {
+        if (passwordComponent.HasPassword())
+        {
+            await RefreshSignatures();
+        }
+        else
+        {
+            passwordComponent.ShowPassword(async () =>
+            {
+                await RefreshSignatures();
+            });
+        }
+    }
+
+    private async Task RefreshSignatures()
+    {
+        if (isCooldown)
+            return;
+        isLoading = true;
+        StateHasChanged();
+        isCooldown = true;
+        await FetchPendingSignatures(FounderProject);
+
+
+        await FetchSignatures();
+        await Task.Delay(1000);
+        
+        isLoading = false;
```

Comment:
Here you wanna put the bool assignments inside a `finally` look at the method `RefreshBalance` in `Wallet.razor`